randovania/gui/lib/multiplayer_session_api.py

```diff
@@ -26,11 +28,11 @@
 OriginalFunc = typing.Callable[Param, RetType]
 
 
-def handle_network_errors[**Param, RetType](
-    fn: typing.Callable[typing.Concatenate[MultiplayerSessionApi, Param], RetType],
-) -> typing.Callable[Param, RetType]:
+def handle_network_errors[T, **P](
+    fn: Callable[P, Coroutine[Any, Any, T]],
+) -> Callable[P, Coroutine[Any, Any, T | None]]:
```

Comment:
https://github.com/randovania/randovania/blob/76f51f214ed5dd9f34131f18bab878b48005dcb6/randovania/server/server_app.py#L51

maybe this should be pulled out into something in `lib` so that it can be reused wherever needed? I find it a lot easier to read